samples/assistant/README.md

```diff
@@ -0,0 +1,221 @@
+# Assistants
+
+This sample demonstrates how to build an AI assistant chatbot with custom skills using Azure Functions and a local build of the experimental OpenAI extension.
+It builds upon the concepts [chatbot](../chatbot) sample, which demonstrates how to create a simple chatbot.
+
+## Introduction
+
+AI assistants are chat bots that can be configured with custom skills. Custom skills are implemented using the `assistantSkillTrigger` binding.
+Custom skills are useful ways to extend the functionality of an AI assistant. For example, you can create a custom skill that saves a todo item to a database, or a custom skill that queries a database for a list of todo items.
+In both cases, the skills are defined in your app and any the language model doesn't need to know anything about the skill implementation, making it useful for interacting with private data.
+
+This OpenAI extension internally uses the [function calling](https://platform.openai.com/docs/guides/function-calling) functionality available in
+`gpt-3.5-turbo` and `gpt-4` models to implement assistant skills.
+
+## Defining skills
+
+You can define a skill by creating a function that uses the `AssistantSkillTrigger` binding. The following C# in-proc example shows a skill that adds a todo item to a database:
```

Comment:
Is it possible to make this work with OOP in this PR or would we want to do this in another PR?

Reply:
Unfortunately, more design work will be required for all out-of-proc languages since we don't have any way to automatically describe the function signatures in those languages. Let's create a separate work item for supporting out-of-process languages, including .NET Isolated.